src/Uno.UWP/Media/Core/MediaSource.cs

```diff
@@ -1,18 +1,29 @@
 using System;
 using Windows.Media.Playback;
+using Windows.Storage.Streams;
 
 namespace Windows.Media.Core
 {
 	public partial class MediaSource : IDisposable, IMediaPlaybackSource
 	{
 		public Uri Uri { get; private set; }
 
+		public IRandomAccessStream Stream { get; private set; }
+
 		public static MediaSource CreateFromUri(Uri uri)
 		{
 			return new MediaSource()
 			{
 				Uri = uri
 			};
 		}
+
+		public static MediaSource CreateFromStream(IRandomAccessStream stream, string _)
+		{
+			return new MediaSource()
+			{
+				Stream = stream
+			};
+		}
```

Comment:
Does this work only for Android? If so, consider adding `#if __ANDROID__`

Reply:
The `#if __ANDROID__` will also fix the build errors in CI

Reply:
for this pr yes. I'm currently implementing it for skia. 

I will add the if. thanks :)